src/main/scala/com/cognite/sdk/scala/v1/dataTypes.scala

```diff
@@ -10,4 +10,13 @@ sealed trait CogniteId
 final case class CogniteExternalId(externalId: String) extends CogniteId
 final case class CogniteInternalId(id: Long) extends CogniteId
 
-final case class TimeRange(min: Instant, max: Instant)
+// min and max needs to optional, since one of them can be provided alone.
+final case class TimeRange(min: Option[Instant] = None, max: Option[Instant] = None)
```

Comment:
This is a good idea, but we should make sure to add a test to see if it works as we think (does our JSON encoding remove the min field if it's set to `None` for example? I don't remember exactly.)

Reply:
I tried this in ammonite and it worked. Should I write a test case testing this specifically?

Reply:
No it's cool :)

Reply:
Actually, can you test what happens if both are none?

Reply:
c.events.filter(EventsFilter(startTime=Option(TimeRange()))).compile.toList

for example, this returns all the events with startTime which are not None. So, it looks ok I guess.